docs/io/fits/usage/cloud.rst

```diff
@@ -0,0 +1,268 @@
+a secret access key in this case. You can use the ``fsspec_kwargs`` parameter
+to provide your key as follows:
+
+.. doctest-skip::
+
+    >>> fsspec_kwargs = {"key": "YOUR-SECRET-KEY-ID",
+    ...                  "secret": "YOUR-SECRET-KEY"}
+    >>> with fits.open(s3_uri, use_fsspec=True, fsspec_kwargs=fsspec_kwargs) as hdul:
+    ...     cutout = hdul[2].section[10:20, 30:50]
+
+.. warning::
+
+    Including secret access keys inside Python code is dangerous because you
+    may accidentally end up revealing your keys when you share your code with
+    others. A better practice is to store your access keys via a configuration
+    file or environment variables. See the ``s3fs`` documentation for guidance.
+
+
+Using :class:`~astropy.nddata.Cutout2D` with cloud-hosted FITS files
+====================================================================
+
+The examples above used the `ImageHDU.section` feature to download
+small cutouts given a set of pixel coordinates. For astronomical images it is
+often more convenient to obtain cutouts based on a sky position and angular
+size rather than array coordinates. For this reason, Astropy provides the
+`astropy.nddata.Cutout2D` tool which makes it easy to obtain cutouts informed
+by an image's World Coordinate System (`~astropy.wcs.WCS`).
+
+This cutout tool can be used in combination with ``fsspec`` and ``.section``.
+For example, assume you happen to know that the image we opened above contains
+a nice edge-on galaxy at the following position::
+
+    >>> # Approximate location of an edge-on galaxy
+    >>> from astropy.coordinates import SkyCoord
+    >>> position = SkyCoord('10h01m41.13s 02d25m20.58s')
+
+We also know that the radius of the galaxy is approximately 5 arcseconds::
+
+    >>> # Approximate size of the galaxy
+    >>> from astropy import units as u
+    >>> size = 5*u.arcsec
+
+Given this sky position and radius, we can use `~astropy.nddata.Cutout2D`
+in combination with ``use_fsspec=True`` and ``.section`` as follows:
+
+.. doctest-requires:: fsspec
+
+    >>> from astropy.nddata import Cutout2D
+    >>> from astropy.wcs import WCS
+    ...
+    >>> with fits.open(s3_uri, use_fsspec=True) as hdul:  # doctest: +REMOTE_DATA
+    ...     wcs = WCS(hdul[1].header)
+    ...     cutout = Cutout2D(hdul[1].section,  # use `.section` rather than `.data`!
+    ...                       position=position,
+    ...                       size=size,
+    ...                       wcs=wcs)
+
+As a final step, you can plot the cutout using Matplotlib as follows:
+
+.. doctest-requires:: fsspec
+
+    >>> import matplotlib.pyplot as plt
+    >>> from astropy.visualization import astropy_mpl_style
+    ...
+    >>> plt.style.use(astropy_mpl_style)  # doctest: +REMOTE_DATA
+    >>> plt.figure()  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+    >>> plt.imshow(cutout.data, cmap='gray')  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+    >>> plt.colorbar()  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
```

Comment:
Use the directive rather than repeated in-line. I think using the directive for the other examples would be a cleaner logic, too, but all of those have only one in-line REMOTE_DATA atm, so I wouldn't insist on changing them (in fact this whole RST document could use a whole file `remote-data` directive, but afaik it's not yet available (https://github.com/astropy/pytest-doctestplus/issues/146)

```suggestion
.. doctest-requires:: fsspec
.. doctest-remote-data::

    >>> import matplotlib.pyplot as plt
    >>> from astropy.visualization import astropy_mpl_style
    ...
    >>> plt.style.use(astropy_mpl_style)
    >>> plt.figure()  # doctest: +IGNORE_OUTPUT
    >>> plt.imshow(cutout.data, cmap='gray')  # doctest: +IGNORE_OUTPUT
    >>> plt.colorbar()  # doctest: +IGNORE_OUTPUT
```

Reply:
For some reason, adding `.. doctest-remote-data::` makes `tox -e build_docs` fail with the following obscure message:

```
reading sources... [ 89%] io/fits/usage/cloud

Exception occurred:
  File "/Users/gb/dev/astropy/.tox/build_docs/lib/python3.9/site-packages/docutils/statemachine.py", line 1156, in __getitem__
    return self.data[i]
IndexError: list index out of range
The full traceback has been saved in /var/folders/x4/y38zs5fj4szf7xj3354kv74m0000gn/T/sphinx-err-p77c2iog.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
ERROR: InvocationError for command /Users/gb/dev/astropy/.tox/build_docs/bin/sphinx-build -b html . _build/html -j auto (exited with code 2)
___________________________________________________________________ summary ____________________________________________________________________
ERROR:   build_docs: commands failed
```

Reply:
interesting, let's move this suggestion out of scope of this PR then. 
could you file a bug issue for it in `pytest-doctestplus`?